webapp-django/crashstats/topcrashers/tests/test_views.py

```diff
@@ -0,0 +1,223 @@
+import mock
+import pyquery
+from nose.tools import eq_, ok_
+
+from django.core.urlresolvers import reverse
+
+from waffle.models import Switch
+
+from crashstats.crashstats.tests.test_views import (
+    BaseTestViews, Response, mocked_post_123
+)
+from crashstats.supersearch.tests.common import (
+    SUPERSEARCH_FIELDS_MOCKED_RESULTS,
+)
+
+
+class TestViews(BaseTestViews):
+    base_url = reverse('topcrashers:topcrashers')
+
+    @staticmethod
+    def setUpClass():
+        TestViews.switch = Switch.objects.create(
+            name='topcrashers-real-time',
+            active=True,
+        )
+
+    @staticmethod
+    def tearDownClass():
+        TestViews.switch.delete()
+
+    @mock.patch('socorro.external.es.supersearch.SuperSearch')
+    @mock.patch('crashstats.crashstats.models.Bugs.get')
+    @mock.patch('requests.get')
+    def test_topcrashers(self, rget, rpost, supersearch):
+
+        def mocked_post(**options):
+            return {
+                "hits": [
+                    {"id": 123456789,
+                     "signature": "Something"},
+                    {"id": 22222,
+                     "signature": u"FakeSignature1 \u7684 Japanese"},
+                    {"id": 33333,
+                     "signature": u"FakeSignature1 \u7684 Japanese"}
+                ]
+            }
+        rpost.side_effect = mocked_post
+
+        def mocked_get(url, params, **options):
+            assert 'supersearch' in url
+
+            if 'supersearch/fields' in url:
+                return Response(SUPERSEARCH_FIELDS_MOCKED_RESULTS)
```

Comment:
One of us will have to change this when supersearchfields are served straight outta socorro.external.es :)